src/Reporter.h

```diff
@@ -144,21 +217,53 @@ class Reporter
 
 	// Syslog a message. This methods does nothing if we're running
 	// offline from a trace.
-	void Syslog(const char* fmt, ...) FMT_ATTR;
+	template <typename... Args> void Syslog(const char* fmt, Args&&... args)
+		{
+		if ( run_state::reading_traces )
+			return;
+
+		char* buf;
+#ifdef HAVE_ASPRINTF
+		asprintf(&buf, fmt, std::forward<Args>(args)...);
+#else
+		util::asprintf(&buf, fmt, std::forward<Args>(args)...);
+#endif
```

Comment:
Just as a thought in passing while this is still in flux: the conditionally compiled code could also be in `utitl::asprintf` so this code could unconditionally call it without having to worry about wrappers or not; in that case `util::asprintf` would also be defined unconditionally.

One benefit of that approach would be that we'd end up with a very simple guideline for reviews, `asprintf` should never be called, unless you are implementing `asprintf`.

Reply:
I left it like this because `util::asprintf` uses `vasprintf`, which requires a va_list to work, which in turn requires `util::asprintf` to be a C-style variadic function. I could make two versions of the function conditionally wrapped in `#ifdef` block, but I don't see a way to call `vasprintf` otherwise.

Reply:
The more correct fix here would be to implement `vasprintf` in libunistd, so it's only Windows that's missing it.